### PR TITLE
Add preview support for several script languages

### LINKED ISSUE #43: Add support for C# scripts
Was just testing the extension and it looks like when adding a new function it add for js only. Would be good to see it also support C#.

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
We've had requests to add support for C# Scripts. ~~Unfortunately, the func cli doesn't support local debugging for C# Scripts (AFAIK)~~. However, we can still provide the following:
1. 'Create New Project': It provides a 'tasks.json' file with the 'Run Functions Host' task, but no 'launch.json' file. (Users must change their `azureFunctions.ProjectLangauge` setting for this support.)
1. 'Create New Function': Most of this support already existed - the only addition is C# Scripts
1. Deploy: Works for all script languages

I added unit tests for each language, but only felt the need to do one function per language (since those could change and I don't think we need to keep our unit tests that up-to-date for preview languages).

Fixes #43 

Update: The func cli _does_ support debugging for C# Scripts. This PR includes now includes a 'launch.json' file for C# Scripts.

### CODE REVIEW

File: src/ProjectSettings.ts [hunk 1]
```diff
@@ -90,15 +91,18 @@ export async function selectTemplateFilter(ui: IUserInterface): Promise<Template
     return <TemplateFilter>result;
 }
 
+export function tryGetLanguageSetting(): ProjectLanguage | undefined {
```
Comment (nturinski) [inline src/ProjectSettings.ts]: You have three functions that getConfiguration based off extensionPrefix and the setting name.  Is it time to create a getWorkspaceSetting function?

File: src/commands/createNewProject/CSharpScriptProjectCreator.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public getLaunchJson(): {} {
+        return {
+            version: '0.2.0',
+            configurations: [
+                {
+                    name: localize('azFunc.attachToNetCoreFunc', "Attach to .NET Core Functions"),
+                    type: 'coreclr',
+                    request: 'attach',
```
Comment (nturinski) [inline src/commands/createNewProject/CSharpScriptProjectCreator.ts]: I'm probably doing something incorrectly, butt seems like C# Scripts attach, but I can't set breakpoints.

File: src/ProjectSettings.ts (hunk 1 above)
Comment (nturinski) [reply src/ProjectSettings.ts]: https://github.com/Microsoft/vscode-azurefunctions/pull/177/files#diff-67685aa049a8b0e5dfd64a65893c8717R94 😆

Comment (EricJizbaMSFT) [reply src/ProjectSettings.ts]: Great minds think alike

File: src/commands/createNewProject/CSharpScriptProjectCreator.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/createNewProject/CSharpScriptProjectCreator.ts]: It was working for me. I'll look into separately
https://github.com/Microsoft/vscode-azurefunctions/issues/180

**Pull request merged**